c_sharp_json_diff/JsonDiff.cs

```diff
@@ -1,55 +1,63 @@
-﻿using Newtonsoft.Json;
+﻿using JsonDiffPatchDotNet;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 using System;
 using System.Collections;
 using System.Collections.Generic;
 
 namespace c_sharp_json_diff
 {
-    public class Result {
-        public List<Diff> Diffs { get; set; } = new List<Diff>();
-
-    }
-    public class Diff
-    {
-        [JsonProperty("property")]
-        public string jsonProperty { get; set; } = "";
-        [JsonProperty("previous")]
-        public double previous { get; set; }
-        [JsonProperty("current")]
-        public double current { get; set; }
-        [JsonProperty("difference")]
-        public double difference { get; set; }
-    }
     public class JsonDiff
     {
-        public static Result GetDiff(IDictionary<string, double> first, IDictionary<string, double> second)
+        public static JObject NumericDiff(string left, string right)
         {
-           ICollection<String> firstKeys = first.Keys;
-            ICollection<String> secondKeys = second.Keys;
-            List<String> matches = new List<string>();
-            Result result = new Result();
-            foreach (string key in secondKeys)
-            {
-                if (firstKeys.Contains(key))
-                {
-                    matches.Add(key);
-                }
-            }
-            foreach(string key in matches)
+            string diff = JsonDiffBetweenTwo(left, right);
+            JObject diffJsonObject = JObject.Parse(diff);
+            foreach(KeyValuePair<string, JToken?> prop in diffJsonObject)
             {
-                first.TryGetValue(key, out double firstValue);
-                second.TryGetValue(key, out double secondValue);
-                if(firstValue != secondValue)
+                if(prop.Value != null)
                 {
-                    Diff item = new Diff();
-                    item.jsonProperty = key;
-                    item.previous = firstValue;
-                    item.current = secondValue;
-                    item.difference = Math.Abs(firstValue - secondValue);
-                    result.Diffs.Add(item);
+                    JArray valuesAsArray = (JArray)prop.Value;
```

Comment:
Rather than doing an unsafe cast from ``JToken`` to ``JArray``, you can safely filter out any non-array fields in the if statement above. Something like this:

``if(prop.Value != null && prop.Value is JArray valuesAsArray)``

---
**Sema Reaction:** :hammer_and_wrench: This code needs a fix